### PR TITLE
Feature/miner

### PR DESCRIPTION
**Pull request by PedroTerriquez**
- . A factory was created to manage the addresses and their information
- . A module was created to store some important methods to convert data (Time, GHs, & ETH)
- . A view was added to show all the information in a table with a nav-tabs.

### CODE REVIEW

File: app/controllers/addresses_controller.rb [hunk 1]
```diff
@@ -11,7 +11,9 @@ def create
 
   def show
     findit
```
Comment (cesargomez89) [inline app/controllers/addresses_controller.rb]: rename this to `find_address` and use a `before_action`

Comment (PedroTerriquez) [reply app/controllers/addresses_controller.rb]: Done and working in the refactor OOD.

File: app/controllers/addresses_controller.rb [hunk 2]
```diff
@@ -10,16 +12,18 @@ def create
...
-    client = EtherClient.new(@address.address)
+    find_address
+    @client = EtherClient.new(@address.address)
+    @min_factory = MinerFactory.new(@client.miner)
+    size
   end
 
   def edit
-    findit
+    find_address
```
Comment (cesargomez89) [inline app/controllers/addresses_controller.rb]: you don't need to add this if it is a callback right?

File: app/controllers/addresses_controller.rb [hunk 3]
```diff
@@ -10,16 +12,14 @@ def create
   end
 
   def show
-    findit
     client = EtherClient.new(@address.address)
+    @min_factory = MinerFactory.new(client.miner)
+    get_miner_information
```
Comment (lacf95) [inline app/controllers/addresses_controller.rb]: Use miner_information instead

File: app/controllers/addresses_controller.rb [hunk 4]
```diff
@@ -28,18 +28,25 @@ def update
   end
 
   def destroy
-    findit
     redirect_to home_index_path unless !@address.destroy
```
Comment (lacf95) [inline app/controllers/addresses_controller.rb]: if @address.destroy

Comment (PedroTerriquez) [reply app/controllers/addresses_controller.rb]: Flash messages?

File: app/factories/miner/miner_statistics.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
+class MinerStatistics
+  include EthermineConvertions
+  def initialize(miner)
```
Comment (lacf95) [inline app/factories/miner/miner_statistics.rb]: miner? statistics attributes instead

File: app/controllers/addresses_controller.rb [hunk 6]
```diff
@@ -1,45 +1,66 @@
...
   def new
     @address = Address.new
   end
 
   def create
     @address = Address.new(address_params)
     @address.user_id = session[:user_id]
-    redirect_to home_index_path unless !@address.save
+    if @address.save
+      flash[:notice] = 'The address was succesfully created'
```
Comment (lacf95) [inline app/controllers/addresses_controller.rb]: Why use flash? access the address errors in the view

File: app/controllers/addresses_controller.rb [hunk 7]
```diff
@@ -1,45 +1,66 @@
...
-  def edit
-    findit
-  end
-  
+  def edit; end
+
   def update
-    findit
     if @address.update(address_params)
+      flash[:notice] = 'The address was succesfully updated'
```
Comment (lacf95) [inline app/controllers/addresses_controller.rb]: Again

File: app/controllers/addresses_controller.rb (hunk 6 above)
Comment (PedroTerriquez) [reply app/controllers/addresses_controller.rb]: That is not an error

File: app/controllers/addresses_controller.rb [hunk 8]
```diff
@@ -1,45 +1,66 @@
...
-      redirect_to 'edit'
+      flash[:error] = @address.errors.full_messages.to_sentence
+      redirect_to edit_address_path @address
     end
   end
 
   def destroy
-    findit
-    redirect_to home_index_path unless !@address.destroy
+    if @address.destroy
```
Comment (lacf95) [inline app/controllers/addresses_controller.rb:39]: And again...

File: app/controllers/addresses_controller.rb (hunk 6 above)
Comment (lacf95) [reply app/controllers/addresses_controller.rb]: It's not, but it smells bad

**Review by lacf95 — CHANGES_REQUESTED** [review]
Only one suggestion

File: app/controllers/addresses_controller.rb (hunk 7 above)
Comment (PedroTerriquez) [reply app/controllers/addresses_controller.rb]: Again is not an error

**Review by lacf95 — APPROVED** [review]

File: app/controllers/addresses_controller.rb (hunk 6 above)
Comment (cesargomez89) [reply app/controllers/addresses_controller.rb]: you can iterate flashes in the view with `flash.each` depending of the kind of error you can assign a class to the message using bootstrap https://v4-alpha.getbootstrap.com/components/alerts/

File: app/views/addresses/new.html.haml [hunk 9]
```diff
@@ -10,4 +10,12 @@
         .form-group
           = form.label :alias
           = form.text_field :alias, class: 'form-control', placeholder: 'Alias for this address', type: 'text'
+        - flash.each do |name,msg|
+          .alert.alert-danger{role: 'alert'}
```
Comment (cesargomez89) [inline app/views/addresses/new.html.haml:14]: nice 👍

File: app/factories/miner_factory.rb [hunk 10]
```diff
@@ -0,0 +1,50 @@
+class MinerFactory
+  load 'ethermine_convertions.rb'
```
Comment (cesargomez89) [inline app/factories/miner_factory.rb]: you don't need to load stuff, rails do it for you

Comment (PedroTerriquez) [reply app/factories/miner_factory.rb]: But if i delete that lines doesnt work, and appear the problem that i send u in slack:

uninitialized constant MinerFactory::MinerSettings

And that for each class.

Comment (cesargomez89) [reply app/factories/miner_factory.rb]: you can try 2 ways
- invoke the class from root `::MinerSettings`
- inherit the nested classes in the module like that `MinerFactory::MinerSettings`
- and the not recommended way, add the path in the application.rb to aoutload the folder

**Pull request merged**